src/App.js

```diff
@@ -1,23 +1,15 @@
-import logo from './logo.svg';
+import { Route, Routes } from 'react-router-dom';
+import WeatherMain from './components/weatherMain';
+import WeatherDetail from './components/weatherDetail';
 import './App.css';
 
 function App() {
```

Comment:
Hi @BilalLiaquat7,

I noticed that your project lacks tests, both unit tests for individual components and integration tests using React Testing Library. According to the project requirements, tests are a crucial part of the submission and should not be overlooked.

Testing is a key aspect of production-grade code and contributes to its maintainability and robustness. Specifically, the requirements indicate that you should:

- Have unit tests for several components of the app.
- Include some pure functions in your Redux code and write tests for them.
- Perform integration tests using React Testing Library.
- Be capable of mocking data fetching and Redux Store connection.

Without tests, it is difficult to assess the reliability of the application. Please add the relevant tests to your project as specified in the requirements, and resubmit your project for another review.

If you have any questions about writing tests or need clarification on the project requirements, feel free to reach out. 

Best regards,
Temitope